src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.WriteCore.cs

```diff
@@ -44,6 +47,18 @@ public partial class JsonConverter<T>
         {
             try
             {
+                Debug.Assert(writer != null);
+
+                if (
+#if NETCOREAPP
+                    !typeof(T).IsValueType &&
```

Comment:
should you:

```csharp
#else
                !IsValueType)
#endif
``

Reply:
btw isn't the only diff between `IsValueType` and `default(T) is not null` `Nullable<T>`? Since nullable can't be polimorphic should we just check for latter?

Reply:
> should you:

Not really needed, `CanBePolymorphic` implies `IsValueType` so it's a redundant check. The only reason it is there is for the JIT to completely eliminate the branch for struct types.

Reply:
> btw isn't the only diff between IsValueType and default(T) is not null Nullable<T>? Since nullable can't be polimorphic should we just check for latter?

We just need to check if `T` is a reference type, so `!typeof(T).IsValueType` is the appropriate logic.